string blob
float32[]  data
